### PR TITLE
Feature/cleanup u3v class

### PR DESCRIPTION
**Pull request by Fixstars-momoko**
Compress the constructor of U3V and U3VGenDC

### CODE REVIEW

File: src/bb/image-io/rt_u3v.h [hunk 1]
```diff
@@ -501,6 +517,344 @@ class U3V {
...
+        for (auto i = 0; i < num_sensor_; ++i) {
+            int32_t num_input_buffer;
+            arv_stream_get_n_buffers(devices_[i].stream_, &num_input_buffer, &(N_output_buffers[i]));
+        }
+        // if all stream has N output buffers, discard N-1 of them
+        for (auto i = 0; i < num_sensor_; ++i) {
+            for (auto j = 0; j < N_output_buffers[i] - 1; ++j) {
+                bufs[i] = arv_stream_timeout_pop_buffer(devices_[i].stream_, timeout_us);
+                if (bufs[i] == nullptr) {
+                    log::error("pop_buffer(L2) failed due to timeout ({}s)", timeout_us * 1e-6f);
```
Comment (xinyuli1204) [inline src/bb/image-io/rt_u3v.h:841]: can you make pop_buffer(L2) failed due to timeout ({}s to pop_buffer failed due to timeout ({}s when consuming all old buffers

File: src/bb/image-io/rt_u3v.h [hunk 2]
```diff
@@ -484,6 +484,22 @@ class U3V {
         return err_;
     }
 
+    void ValidateUserInput(int32_t num_detected_device, char* dev_id){
```
Comment (iitaku) [inline src/bb/image-io/rt_u3v.h:487]: Let's unify the naming convention. In ion-kit, we use `snake_case` for the function and variable name.

File: src/bb/image-io/rt_u3v.h [hunk 3]
```diff
@@ -484,6 +484,22 @@ class U3V {
...
+        if (num_detected_device > num_sensor_ && dev_id == nullptr) {
+            log::info("{} devices are found; The first {} device is selected", num_detected_device, num_sensor_);
+        }
+
+        log::info("Creating U3V instance with {} sensors...", num_sensor_);
+        log::info("Acquisition option::{} is {}", "frame_sync_", frame_sync_);
+        log::info("Acquisition option::{} is {}", "realtime_display_mode_", realtime_display_mode_);
+    }
+
     GError* CommandAcquisitionModeContdStart(){
```
Comment (iitaku) [inline src/bb/image-io/rt_u3v.h:503]: Do we have a specific policy for the error handling? It looks function is expected to return GError *, but it actually throws an exception in many cases.

File: src/bb/image-io/rt_u3v.h (hunk 2 above)
Comment (Fixstars-momoko) [reply src/bb/image-io/rt_u3v.h]: solved in [0a63a5b](https://github.com/fixstars/ion-kit/pull/324/commits/0a63a5bd39dd9c21526c6edadbc28402d3ee6e5e)

File: src/bb/image-io/rt_u3v.h (hunk 3 above)
Comment (Fixstars-momoko) [reply src/bb/image-io/rt_u3v.h]: Updated in [c654c02](https://github.com/fixstars/ion-kit/pull/324/commits/c654c02858f79ed72af1cf0b998e98b59f101495)

**Pull request merged**